packages/@aws-cdk/aws-lambda-golang/lib/Dockerfile

```diff
@@ -0,0 +1,12 @@
+# The correct AWS SAM build image based on the runtime of the function will be
+# passed as build arg. The default allows to do `docker build .` when testing.
+ARG IMAGE=lambci/lambda:build-go1.x
```

Comment:
Let's use the official images from the public ECR repo - e.g. https://gallery.ecr.aws/sam/build-provided

Reply:
I'm using the image specified in the `lambda.Runtime.GO_1_X.bundlingDockerImage`

https://github.com/aws/aws-cdk/blob/b4956ea009291ab64bc58012e2108fb6f201c2ca/packages/%40aws-cdk/aws-lambda/lib/runtime.ts#L161-L163

The SAM team has not created bundling docker images for Go yet, and based on the docs doesn't actually support building Go functions in a container (only local build supported). The only Go image on the public ECR is the [bitnami golang image](https://gallery.ecr.aws/bitnami/golang). I'm using this in the integration test so I could update it to this, but then we should probably update the `lambda.Runtime.GO_1_X.bundlingDockerImage` as well.